include/lorawan/lorawan.h

```diff
@@ -202,13 +201,12 @@ int lorawan_start(void);
  * @param len        Length of the buffer to be sent. Maximum length of this
  *                   buffer is 255 bytes but the actual payload size varies with
  *                   region and datarate.
- * @param flags      Flag used to determine the type of message being sent. It
- *                   could be one of the lorawan_message_type. The default
- *                   behaviour is unconfirmed message.
+ * @param type       Specifies if the message shall be confirmed or unconfirmed.
+ *                   Must be one of lorawan_message_type.
  *
  * @return 0 if successful, negative errno code if failure
  */
-int lorawan_send(uint8_t port, uint8_t *data, uint8_t len, uint8_t flags);
+int lorawan_send(uint8_t port, uint8_t *data, uint8_t len, uint8_t type);
```

Comment:
If you're changing `could be one of the lorawan_message_type` to `Must be one of lorawan_message_type` then you may as well make the type parameter explicit `enum lorawan_message_type type`

Reply:
The idea was that existing code doesn't break, but everyone who reads the docs does it correct.

I'm also fine with `enum lorawan_message_type`. Let's see what others think.

Reply:
During the initial API review we decided not to define a type for unconfirmed message as it is the default. But I think defining it also doesn't hurt but then please change the datatype of `type` to enum.

Reply:
Oh wait... Please let the `flags` parameter stay as it is. We may want to add some flags in the future.

Reply:
> Are there any other types than confirmed and unconfirmed?
> 
> If we want to use these flags for something else than the message type in the future, the enum should be called `lorawan_message_flags` instead. Otherwise there is no relation to the different bits that can be set as flags.

After going back and forth, I came to the conclusion that we should be just using `lorawan_message_type`. So please change the datatype of the `type` parameter.